### DISCUSSION TITLE
Which versions of Python are supported and how many versions will be supported at once?

### DISCUSSION THREAD

**Discussion by joaogui1** [jax-ml/jax#6005] [Q&A]
Elaborating on the second question, do you plan to support a sliding window of python versions? (Like 3.7, 3.8, 3.9 -> 3.8, 3.9, 3.10)

**Comment by hawkinsp** [comment]
We don't have a formal policy for this, but informally: "we will match NumPy's policy". That is, we aim to support at least the set of versions that NumPy supports: https://numpy.org/neps/nep-0029-deprecation_policy.html

Currently we support 3.6 - 3.9. Strictly speaking, we should have dropped Python 3.6 support already, and we will probably do that as soon as some of our key users are fully off Python 3.6.

One reason we are eager to drop Python 3.6 support is because JAX gives much better stack traces in Python 3.7!

**Reply by joaogui1** [reply]
Makes a lot of sense, thanks for the answer!
